kustomize/commands/edit/add/addgenerator.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"log"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/util"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addGeneratorOptions struct {
+	generatorFilePaths []string
+}
+
+// newCmdAddGenerator adds the name of a file containing a generator
+// configuration to the kustomization file.
+func newCmdAddGenerator(fSys filesys.FileSystem) *cobra.Command {
+	var o addGeneratorOptions
+
+	cmd := &cobra.Command{
+		Use:   "generator",
+		Short: "Add the name of a file containing a generator configuration to the kustomization file.",
+		Example: `
+		add generator {filepath}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+			return o.RunAddGenerator(fSys)
+		},
+	}
+	return cmd
+}
+
+// Validate validates add generator command.
+func (o *addGeneratorOptions) Validate(args []string) error {
```

Comment:
Suggest to add validation to make sure

- The existence of the path. Reason:  if not checked, `kustomize build` will fail for the same reason anyway. it makes `kustomize edit` feel less reliable/stable since it edits user files without necessary validation checks even the same logic is used in `kustomize build`.   
- Format of the generator.  Reason:  whoever uses `kustomize edit` is more expecting kustomize to give guidance and validation checks. Thus, validation checks like valid yaml, valid OpenApi could always be super helpful. We can add a TODO here if not planning to add the validation check for now. But overall I feel like expanding the validation checks on `kustomize edit`  is a good approach to introduce kustomize to rely more on this command.

Reply:
To your first point, yeah the validation of the path is done in `GlobPatterns`. But I think it makes more sense to do that step in `Validate`, so I've moved it over.v

For the second point, I think that's a good idea but would be better in its own PR, so I added a TODO.

Reply:
sg.